src/util/Monthpicker/Monthpicker.js

```diff
@@ -0,0 +1,291 @@
+
+import {
+  Button,
+  Col,
+  HasCommand,
+  IconButton,
+  Popper,
+  TextField,
+} from '@folio/stripes/components';
+
+import css from './Monthpicker.css';
+
+const Monthpicker = ({
+  backendDateFormat = 'yyyy-MM',
+  dateFormat,
+  input,
+  isRequired,
+  meta,
+  textLabel = '',
+}) => {
+  const [showCalendar, setShowCalendar] = useState(false);
+  const lastValidDateRef = useRef({ year: null, month: null });
+  const container = useRef(null);
+  const intl = useIntl();
+
+  const normalizeLuxonFormat = (format) => {
+    return format
+      .replace(/Y/g, 'y')
+      .replace(/m/g, 'M');
+  };
+
+  const getDateFormatFromLocale = (locale) => {
+    const parts = new Intl.DateTimeFormat(locale, {
+      year: 'numeric',
+      month: '2-digit',
+    }).formatToParts(new Date());
+
+    return parts
+      .map((part) => {
+        if (part.type === 'month') return 'MM';
+        if (part.type === 'year') return 'yyyy';
+        return part.value;
+      })
+      .join('');
+  };
+
+  const resolvedDateFormat = useMemo(() => {
+    const localDate = getDateFormatFromLocale(intl.locale);
+    return normalizeLuxonFormat(dateFormat ?? localDate);
+  }, [dateFormat, intl.locale]);
+
+  const resolvedBackendDateFormat = normalizeLuxonFormat(backendDateFormat);
+
+  const isValidYear = (year) => {
+    const num = Number(year);
+    return Number.isInteger(num) && num >= 1000 && num <= 9999;
+  };
+
+  // const isValidMonth = (month) => {
+  //   const monthIndex = month - 1;
+  //   return !Number.isNaN(monthIndex) && monthIndex >= 0 && monthIndex <= 11;
+  // };
+
+  // const ensureValidMonth = (month) => {
+  //   if (isValidMonth(month)) return month;
+  //   return new Date().getMonth() + 1;
+  // };
+
+  const ensureValidYear = (year) => {
+    if (isValidYear(year)) return year;
+    return new Date().getFullYear();
+  };
+
+  const buildDateString = (year, month, format) => {
+    const dt = DateTime.fromObject({ year: ensureValidYear(year), month });
+    return dt.toFormat(format);
+  };
+
+  useEffect(() => {
+    const backendFormat = normalizeLuxonFormat(backendDateFormat);
+    const dt = DateTime.fromFormat(input?.value, backendFormat);
+
+    if (dt.isValid) {
+      lastValidDateRef.current = {
+        year: dt.year,
+        month: dt.month
+      };
+    } else if (!lastValidDateRef.current.year || !lastValidDateRef.current.month) {
+      const now = DateTime.local();
+      lastValidDateRef.current = { year: now.year, month: now.month };
+    }
+  }, [input?.value, backendDateFormat]);
+
+  const getLocalizedMonthAbbreviations = () => {
+    return Array.from({ length: 12 }, (_, i) => {
+      return new Intl.DateTimeFormat(intl.locale, { month: 'short' }).format(new Date(2000, i, 1));
+    });
+  };
+
+  const convertDateFormat = (inputValue, fromFormat, toFormat) => {
+    const dt = DateTime.fromFormat(inputValue, fromFormat);
+    return dt.isValid ? dt.toFormat(toFormat) : null;
```

Comment:
Use `''` instead of `null` as tests yield a warning:

```
Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
```